src/radical/pilot/agent/executing/popen.py

```diff
@@ -68,7 +68,8 @@ def initialize(self):
     #
     def get_task(self, tid):
 
-        return self._tasks.get(tid)
+        with self._check_lock:
```

Comment:
I don't understand what the lock is achieving here, the `.get()` should be safe either way? If the task was removed just before the lock is set, the result is the same.

Reply:
yeah, agree, brought it back